src/carp.cc

```diff
@@ -123,17 +98,17 @@ carpInit(void)
      * X_k = pow (X_k, {1/(K-k+1)})
      * simplified to have X_1 part of the loop
      */
-    K = n_carp_peers;
+    const auto K = CarpPeers.size();
 
     P_last = 0.0;       /* Empty P_0 */
 
     Xn = 1.0;           /* Empty starting point of X_1 * X_2 * ... * X_{x-1} */
 
     X_last = 0.0;       /* Empty X_0, nullifies the first pow statement */
 
-    for (k = 1; k <= K; ++k) {
+    for (size_t k = 1; k <= K; ++k) {
         double Kk1 = (double) (K - k + 1);
-        p = carp_peers[k - 1];
+        auto p = CarpPeers[k - 1].get();
```

Comment:
If possible:

```suggestion
        const auto p = CarpPeers[k - 1].get();
```

Please check for other, similar cases in modified/new code.